### PR TITLE
Fix exception arguments and documentation

### PR DESCRIPTION
**Pull request by hluk**
Signed-off-by: Lukas Holecek <hluk@email.cz>

### CODE REVIEW

File: fedora_messaging/exceptions.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ class PermissionException(BaseException):
         obj_type (str): The type of object being accessed that caused the
             permission error. May be None if the cause is unknown.
         description (object): The description of the object, if any. May be None.
-        reason (str): The reason the server gave for the permission error, if
+        reason (str): The reason the server gave the permission error, if
```
Comment (pypingou) [inline fedora_messaging/exceptions.py]: The `for` seems justified for me, it's the reason for the error, no?

**Comment by pypingou** [comment]
Looks good to me, though I expect we may want tests with this?

File: fedora_messaging/exceptions.py (hunk 1 above)
Comment (hluk) [reply fedora_messaging/exceptions.py]: I misread it. Sometimes I don't understand English. I'm reverting it.

**Comment by hluk** [comment]
Any idea what's up with the integration test? For me they get also stuck locally.

File: fedora_messaging/exceptions.py [hunk 2]
```diff
@@ -75,12 +75,25 @@ def __str__(self):
...
+        reason (str): The reason the server gave for the publish error.
+    """
 
     def __init__(self, reason=None, **kwargs):
         super(PublishException, self).__init__(**kwargs)
         self.reason = reason
 
+    def __str__(self):
+        if self.reason:
+            return "Publish error: " + self.reason
```
Comment (jeremycline) [inline fedora_messaging/exceptions.py]: It'd be good to call str on self.reason so users can pass in non-text objects (for example, a lower-level exception)

File: fedora_messaging/exceptions.py [hunk 3]
```diff
@@ -106,11 +119,22 @@ class ConnectionException(BaseException):
...
-    def __init__(self, *args, **kwargs):
-        super(ConnectionException, self).__init__(*args)
-        self.reason = kwargs.get("reason")
+    def __init__(self, reason=None, **kwargs):
+        super(ConnectionException, self).__init__(**kwargs)
+        self.reason = reason
+
+    def __str__(self):
+        if self.reason:
+            return "Connection error: " + self.reason
```
Comment (jeremycline) [inline fedora_messaging/exceptions.py]: Same deal here with str

**Comment by jeremycline** [comment]
Actually, current master also passes CI. Maybe one of the tests is expecting something from one of these exceptions that broke? I don't have time to dig into it today, but I might be able to in the next couple days.

**Pull request closed without merging**